pkg/order/mempool/tx_store.go

```diff
@@ -194,6 +204,50 @@ func (nc *nonceCache) setPendingNonce(account string, nonce uint64) {
 	nc.pendingNonces[account] = nonce
 }
 
+func (nc *nonceCache) updatePendingNonce(newPending map[string]uint64) {
+	storageBatch := nc.fallback.NewBatch()
+	defer storageBatch.Commit()
+	for account, pendingNonce := range newPending {
+		nc.setPendingNonce(account, pendingNonce)
+		storageBatch.Put(pendingNonceKey(account), []byte(strconv.FormatUint(pendingNonce, 10)))
+	}
+}
+
+func (nc *nonceCache) updateCommittedNonce(newCommitted map[string]uint64) {
+	storageBatch := nc.fallback.NewBatch()
+	defer storageBatch.Commit()
+	for account, committedNonce := range newCommitted {
+		nc.setCommitNonce(account, committedNonce)
+		storageBatch.Put(committedNonceKey(account), []byte(strconv.FormatUint(committedNonce, 10)))
+		pendingNonce := nc.getPendingNonce(account)
```

Comment:
No longer need to update pendingnonce, because the restart node will be restored pendingnonce from the db